publish-mqtt/src/main.rs

```diff
@@ -483,3 +482,23 @@ async fn handle_bluetooth_event(
 
     Ok(())
 }
+
+/// Read the given file of key-value pairs into a hashmap.
+/// Returns an empty hashmap if the file doesn't exist, or an error if it is malformed.
+pub fn hashmap_from_file(filename: &str) -> Result<HashMap<String, String>, io::Error> {
```

Comment:
(observation) I probably would have put it a bit closer to `run_sensor_system()` (it's the first thing that run_sensor_system() does, so probably immediately below) but in the long run we will probably split this crate out into modules too, so it doesn't really matter where it goes for now.

Reply:
Done.